<a href="https://colab.research.google.com/github/NRJaggers/Applied-Stochastic-Processes/blob/main/PracticeProblems/Poisson_Processes.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Poisson Processes

---

## Prompt

\\Original Prompt can be found [here](https://bookdown.org/kevin_davisross/applied-stochastic-processes/pp-poisson-processes.html). A copy of the prompt  can be found on github under [/PracticeProblems/Prompts](https://github.com/NRJaggers/Applied-Stochastic-Processes/tree/main/PracticeProblems/Prompts) along with completed exercises under [/PracticeProblems](https://github.com/NRJaggers/Applied-Stochastic-Processes/tree/main/PracticeProblems).

---

## Problems

###1.

Poisson process with $\lambda=3$ calls per minute

####a.

$\text{E}(N_1) = \mu = \lambda \cdot t = 3 \cdot 1 = 3$

####b.

$\text{SD}(N_1) = \sqrt{\text{Var}(N_1)} = \sqrt{\mu} = \sqrt{3} $

####c.

$\text{P}(X=x) = e^{-\mu}\frac{\mu^{x}}{x!} \qquad \mu = \lambda \cdot t$ <br> <br>

$\text{P}(N_1=2) = e^{-(3*1)}\frac{(3*1)^{2}}{2!} = 0.224$

In [ ]:
import math

def poisson(rate, time, events):
  #rate = lambda
  mu = rate * time
  return math.exp(-mu)*(mu**(events))/math.factorial(events)

poisson(3,1,2)

0.22404180765538775

####d.

$\text{P}(N_1=2, N_3=6) = \text{P}(N_1=2, (N_3-N_1)=4)$

Once disjoint, they are independent and can be separated.

$\text{P}(N_1=2) * \text{P}((N_3-N_1)=4)$

The second term can be rewritten as a probability with different RV but the same distribution.

$\text{P}((N_3-N_1)=4) = \text{P}(N_2=4)$

The probability we solve for results in:

$\text{P}(N_1=2) * \text{P}(N_2=4) = (0.224) * (0.134) = 0.0300$

In [ ]:
p1 = poisson(3,1,2)
p2 = poisson(3,2,4)
p1*p2

0.029988582393063134

####e.

$\text{P}(N_1=2|N_3=6)$

There are a few ways to go about this. First we will rewrite and split up the conditional probability.

$\text{P}(N_1=2|N_3=6) = \frac{\text{P}(N_1=2,N_3=6)}{\text{P}(N_3=6)} = \frac{\text{P}(N_1=2,(N_3-N_1)=4)}{\text{P}(N_3=6)} = \frac{\text{P}(N_1=2)*\text{P}(N_2=4)}{\text{P}(N_3=6)} = \frac{(0.224)*(0.134)}{(0.091)} = 0.3298$

The second is that we can use properties of the Poisson Process and rewrite this as a calculation using the binomial distribution.

$P(X = k) = \binom{n}{k} \cdot p^k \cdot (1 - p)^{n - k}$

Where n is the total events from the given, k is the events from the probability of interest, and p is the ratio of times.

$P(X = 2) = \binom{6}{2} \cdot (1/3)^2 \cdot (1 - (1/3))^{6 - 2} = 0.3292$

In [ ]:
p3 = poisson(3,3,6)
p1*p2/p3

0.3292181069958848

In [ ]:
from scipy.stats import binom

n = 6  # number of trials
p = 1/3  # probability of success
k = 2   # number of successes

# Calculate the PMF of the binomial distribution
pmf_value = binom.pmf(k, n, p)

print("PMF value:", pmf_value)


PMF value: 0.3292181069958847


####f.

$\text{P}(N_3=6|N_1=2) = \frac{\text{P}(N_3=6,N_1=2)}{\text{P}(N_1=2)} = \frac{\text{P}((N_3-N_1)=4)*\text{P}(N_1=2)}{\text{P}(N_1=2)} = \text{P}(N_2=4) = 0.134$

In [ ]:
p2

0.13385261753998337

####g.

$\text{E}[N_{15}|N_{10}=8]$

$E[N(15) | N(10) = 8] = E[N(15) - N(10) + N(10) | N(10) = 8]$

$E[N(15) - N(10) | N(10) = 8] + E[N(10) | N(10) = 8]$

$ E[N(10) | N(10) = 8] = 8$

$E[N(15) - N(10) | N(10) = 8] = E[N(15) - N(10)] = E[N(5)]$

$E[N(5)] + 8 = \lambda t +8 = (3)(5)+8 = 23$

####h.

$\text{E}[N_{10}|N_{15}=8]$

you know there is 8 events by 15, so then the distribution becomes uniform over the time period for the events. Therefore:

$\text{E}[N_{10}|N_{15}=8] = \frac{10}{15}*8 = 5.\bar{33}$

####i.

$\text{Cov}(N_{10}, N_{15})$

$\text{Cov}(N_{10}, N_{15}) = \text{Cov}(N_{10}, N_{15}-N_{10}+N_{10})$

$\text{Cov}(N_{10}, N_{15}-N_{10}) + \text{Cov}(N_{10}, N_{10})$

Covariance of two independent things is 0 and convariance of the same variable is variance of that variable.

$0 + \text{Var}(N_{10})$

Mean and Variance of a poisson process are both $\mu  = \lambda t$

$\text{Cov}(N_{10}, N_{15}) = 10t = 10(3) = 30$

####j.

$\text{Corr}(N_{10}, N_{15})$

$\text{Corr}(N_{10}, N_{15}) = \frac{\text{Cov}(N_{10}, N_{15})}{\sqrt{\text{Var}(N_{10})\text{Var}(N_{15})}} = \frac{30}{\sqrt{\lambda^2(10)(15)}} = $

In [4]:
import math
corr = 30/(math.sqrt((3*3)*(10)*(15)))
corr

0.8164965809277259

###2.

Poisson process with mean rate 1.5 spam emails per minute.

$\lambda = 1.5$

####a.

In [ ]:
poisson(1.5,2,4)

0.16803135574154082

$\text{P}(N_2=4) = 0.168$

####b.

might be a better/different way to write this?
otherwise fill in t and solve?

$\text{P}(W_2>3+t|W_1=t-2)$

####c.

In [ ]:
def exponential_gt(rate, wait):
  return math.exp(-rate*wait)

def exponential_lte(rate, wait):
  return 1 - math.exp(-rate*wait)

In [ ]:
exponential_gt(1.5,3)*exponential_lte(1.5,2)

0.010555912168094473

$\text{P}(W>3, W<5) = \text{P}(W<5|W>3)\text{P}(W>3) = \text{P}(W<2)\text{P}(W>3) = 0.011$

should it be like this or are they indepenent so you can separate them fromt the beginning?

####d.

In [ ]:
poisson(1.5,2,1)

0.14936120510359183

$\text{P}((N_5 - N_3) = 1) = \text{P}(N_2 = 1) = 0.149$

####e.

In [ ]:
binom.pmf(1, 1, 1/5)

0.2

In [ ]:
1/5

0.2

$\text{P}(N_1 = 1|N_5 = 1)$

####f.

In [ ]:
binom.pmf(2, 5, 1/3)

0.3292181069958849

$\text{P}(N_1 = 2|N_3 = 5)$

---

## Reflection

With each Practice Problem submission, you should also take a few minutes to reflect on the practice problems and related material and thoughtfully respond to the following.

1. Write a sentence or two summarizing - in your own words - one big idea from this material.
2. Describe one particular aspect of this material that you found challenging and why. Then describe how you dealt with the challenge, or what you can do to deal with the challenge.
3. Write at least one specific question you have about this material. I strongly encourage you to post your questions directly in Discord as that's where I'll respond to questions. That is, if you really want an answer to your question - which I hope you do! - please post the question directly in Discord.

I won't lie, this one kicked my butt a little bit. I feel it was a good practice though and that I got to get my hands dirty with the material. Some of the stuff I was shakey on before definitly feels more solid now. I struggled with 1.g-j and I asked about it in the discord. I also struggled with setting up some of the problems from 2.